p1-staking-pool-dyn/src/lib.rs

```diff
@@ -150,15 +151,15 @@ impl Contract {
         let aid = env::predecessor_account_id();
         let mut vault = self.get_vault_or_default(&aid);
         assert!(
-            vault.staked > 0 && amount <= vault.staked + MIN_STAKE,
+            amount <= vault.staked,
             "Invalid amount, you have not that much staked"
         );
-        if vault.staked >= MIN_STAKE && amount >= vault.staked - MIN_STAKE {
-            //unstake all => close -- simplify UI
```

Comment:
Here is the fix. we want to unstake all if user doesn't have the required deposit - this caused a dead lock when user has less than `MIN_STAKE`.